### ISSUE TITLE
questions regarding the Sobol index implementation

### ISSUE THREAD

**Issue by aufl** [SALib/SALib#620]
Hello, 
I have some questions regarding the Sobol indices. I have been using the functions SALib.sample.sobol.sample() and SALib.analyze.sobol.analyse() functions and I am trying to understand how the implementation in these functions relate to the different estimators that are compared in the paper Puy et al. (2022).

As I understand it from Puy et al., the “Jansen estimator” from Jansen 1999 is the “gold standard” estimator for sobol indices. In the paper Saltelli et al. 2010, which the SALib-documentation of SALib.analyze.sobol.analyse() refers to, the authors also compare some different choices in the implementation of computing sobol indices. There the authors are concluding that the best estimates are obtained by using the Jansens estimator (and quasi-random sampling, radial sampling and number of resample matrices n=1).

The estimator implemented in SALib allows negative estimates (https://github.com/SALib/SALib/issues/102, I also have experienced it for my model), whereas I think the Jansen estimator does not (?). Therefore I was a bit confused about which estimator is actually implemented in SALib. 
 
To sum up, my question is: Which estimators are used for the first order and total order sobol indices (e.g. referring to table 2 in Saltelli et al (2010), or table 1 in Puy et al.( 2022)))? 

Thank you in advance! 

References: 
Puy et al. “A comprehensive comparison of total-order estimators for global sensitivity analysis”, International Journal for Uncertainty Quantification (2022), DOI: 10.1615/Int.J.UncertaintyQuantification.2021038133.

M. Jansen “Analysis of variance designs for model output”, Computer Physics Communications (1999), DOI: 10.1016/S0010-4655(98)00154-4

Saltelli et al. “Variance based sensitivity analysis of model output. Design and estimator for the total sensitivity index”, Computer Physics Communications (2010), DOI: 10.1016/j.cpc.2009.09.018

**Comment by tupui** [comment]
Hi @aufl, thank you for the question.

We are using Saltelli2010 here:

$$ S_i = \frac{1}{N} \sum_{j=1}^N f(\mathbf{B})_j (f(\mathbf{AB}^{(i)})_j - f(\mathbf{A})_j) $$

$$ S_{T_i} = \frac{1}{N} \sum_{j=1}^N (f(\mathbf{A})_j - f(\mathbf{AB}^{(i)})_j)^2 $$

https://github.com/SALib/SALib/blob/da19e0901f9355de02f9691b394154354fbe4792/src/SALib/analyze/sobol.py#L209-L219C12

The negative values are numerical noise. There is not much we can do there besides recommending users to increase the number of samples to perform the calculation.

Hope that helps 😃 Let us know if you have more questions.
